packages/amplify_datastore/lib/method_channel_datastore.dart

```diff
@@ -1,172 +1,208 @@
-      await _channel.invokeMethod('delete', <String, dynamic>{
-        'modelName': model.getInstanceType().modelName(),
-        'serializedModel': model.toJson(),
-      });
-    } on PlatformException catch (e) {
-      throw _deserializeException(e);
-    }
-  }
-
-  @override
-  Future<void> save<T extends Model>(T model) async {
-    try {
-      var methodChannelSaveInput = <String, dynamic>{
-        'modelName': model.getInstanceType().modelName(),
-        'serializedModel': model.toJson(),
-      };
-      await _channel.invokeMethod('save', methodChannelSaveInput);
-    } on PlatformException catch (e) {
-      throw _deserializeException(e);
-    }
-  }
-
-  @override
-  Stream<SubscriptionEvent<T>> observe<T extends Model>(
-      ModelType<T> modelType) {
-    // Step #1. Open the event channel if it's not already open. Note
-    // that there is only one event channel for all observe calls for all models
-    const _eventChannel =
-        EventChannel('com.amazonaws.amplify/datastore_observe_events');
-    _allModelsStreamFromMethodChannel = _allModelsStreamFromMethodChannel ??
-        _eventChannel.receiveBroadcastStream(0);
-
-    // Step #2. Apply client side filtering on the stream.
-    // Currently only modelType filtering is supported.
-    Stream<dynamic> filteredStream =
-        _allModelsStreamFromMethodChannel.where((event) {
-      //TODO: errors are not model specific. Should we pass all errors to users
-      return _getModelNameFromEvent(event) == modelType.modelName();
-    });
-
-    // Step #3. Deserialize events and return new broadcast stream
-    return filteredStream
-        .map((event) => SubscriptionEvent.fromMap(event, modelType))
-        .asBroadcastStream()
-        .cast<SubscriptionEvent<T>>();
-  }
-
-  @override
-  Future<void> clear() async {
-    try {
-      await _channel.invokeMethod('clear');
-    } on PlatformException catch (e) {
-      throw _deserializeException(e);
-    }
-  }
-
-  String _getModelNameFromEvent(Map<dynamic, dynamic> serializedEvent) {
-    Map<String, dynamic> serializedItem =
-        Map<String, dynamic>.from(serializedEvent["item"]);
-    return serializedItem["modelName"] as String;
-  }
-
-  AmplifyException _deserializeException(PlatformException e) {
-    if (e.code == 'DataStoreException') {
-      return DataStoreException.fromMap(Map<String, String>.from(e.details));
-    } else if (e.code == 'AmplifyAlreadyConfiguredException') {
-      return AmplifyAlreadyConfiguredException.fromMap(
-          Map<String, String>.from(e.details));
-    } else {
-      // This shouldn't happen. All exceptions coming from platform for
-      // amplify_datastore should have a known code. Throw an unknown error.
-      return DataStoreException(
-          AmplifyExceptionMessages.missingExceptionMessage,
-          recoverySuggestion:
-              AmplifyExceptionMessages.missingRecoverySuggestion,
-          underlyingException: e.toString());
-    }
-  }
-}
+/*
```

Comment:
Probably worth noting which lines were actually changed since the whole file is being marked for comparison